staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_proxy.go

```diff
@@ -170,6 +190,16 @@ func (r *proxyHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	handler.ServeHTTP(w, newReq)
 }
 
+func (r *proxyHandler) storeExternalProxyInfo(handlingInfo proxyHandlingInfo, serverName string, rt http.RoundTripper) {
+	if handlingInfo.externalServerName == serverName && handlingInfo.externalProxyRoundTripper == rt {
+		return
+	}
+
+	handlingInfo.externalServerName = serverName
+	handlingInfo.externalProxyRoundTripper = rt
+	r.handlingInfo.Store(handlingInfo)
```

Comment:
doing the store here is unsynchronized with the `.Store` in `updateAPIService`.  what prevents a scenario like this?

1. thread/A `ServeHTTP` called
2. thread/A `handlingInfo` for `oldService` loaded
3. thread/B `updateAPIService` called and stores `newService` into `handlingInfo`
4. thread/A calls `storeExternalProxyInfo` with info for `oldService`
5. the `proxyHandler` is now using `oldService` even though it should be  using `newService`

Reply:
discussed in slack, but great catch. i was only considering updates to the `Service`. i'll attempt to address this